src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunnerMode.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.RegularExpressions
+{
+    /// <summary>Represents the mode of execution for a <see cref="RegexRunner"/>.</summary>
+    internal enum RegexRunnerMode
```

Comment:
Of course it is not very relevant for this PR, but I suppose at some point we should consider exposing this type, as well as making the RegexRunner's field with this info protected so that source generated engines can make use of it. The main advantage I see in the future, is once we have NonBacktracking engine supported by the source generator, but perhaps there are other cases where we might consider this. For example, even though it would probably not help a lot, we could decide to omit calls to Capture() and Uncapture() with our current source generated engine in case those are not going to be used.

Reply:
We also don't know when doing the source generation which mode will be used, so we'd still need to emit the Capture/Uncapture-related code, it would just be guarded by emitted checks for that mode.

Reply:
right, I wasn't suggesting not emitting the code, just not executing it by putting it behind and `if`. Might not be very helpful for getting the Captures, but might be a bit helpful for skipping unwinding all captures when a potential match fails after doing several Captures. And yes, even if using isMatch, we would still need to Capture for backreferences and backreference conditionals.